processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -365,7 +365,10 @@ else if ( method.isEnumMapping() ) {
                     .build();
 
                 if ( beanMappingMethod != null ) {
-                    hasFactoryMethod = beanMappingMethod.getFactoryMethod() != null;
+                    // We can consider that the mapping method has a factory method if it has one,
+                    // or if the result type of the methods is not an interface
+                    hasFactoryMethod = beanMappingMethod.getFactoryMethod() != null ||
+                        !beanMappingMethod.getResultType().isInterface();
```

Comment:
Nothing. It works like it has been working for now. We will generate a Mapper and there will be a compilation problem. The issue with #657 is that in line [376](https://github.com/mapstruct/mapstruct/blob/af68fcf6169ffcbee24a45df1f2ef0796a930857/processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java#L376) we explicitly call `reportErrorIfNoImplementationTypeIsRegisteredForInterfaceReturnType` and there we only check the `SourceMethod` and we fail if the return type is Interface and there is no implementation type. 

If we want to make sure that the result type is not an abstract class then we need to change something in the [BeanMappingMethod](https://github.com/mapstruct/mapstruct/blob/af68fcf6169ffcbee24a45df1f2ef0796a930857/processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java#L165-L171) and add a check that the result type is not abstract. 

Should I add the check about an abstract methods to this PR as well?